### PR TITLE
GEOMESA-1134 Adding avro compression and tools output compression

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

Note: stacks on top of #835

### CODE REVIEW

File: geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroDataFileWriter.scala [hunk 1]
```diff
@@ -36,7 +46,7 @@ class AvroDataFileWriter(os: OutputStream, sft: SimpleFeatureType) extends Close
 
   def append(sf: SimpleFeature): Unit =  dfw.append(sf)
 
-  override def close(): Unit = if (dfw != null) dfw.close()
+  override def close(): Unit = if (dfw != null) { dfw.close() }
```
Comment (jahhulbert-ccri) [inline geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroDataFileWriter.scala]: :+1: i know i know

File: geomesa-tools/README.md [hunk 2]
```diff
@@ -412,6 +412,8 @@ To export features, use the `export` command.
         -F, --format
            Format to export (csv|tsv|gml|json|shp|bin)
            Default: csv
+        --gzip
+           level of gzip compression to apply to output, from 1-9
```
Comment (jahhulbert-ccri) [inline geomesa-tools/README.md]: I'm thinking by default we should compress avro at least

File: geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroDataFileWriter.scala [hunk 3]
```diff
@@ -9,23 +9,33 @@
...
+ * Class that writes out Avro SimpleFeature Data Files which carry the SimpleFeatureType Schema
+ * along with them.
+ *
+ * @param os output stream.
+ * @param sft simple feature type being written
+ * @param compression compression level, from -1 to 9. @see java.util.zip.Deflater
+ */
+class AvroDataFileWriter(os: OutputStream,
+                         sft: SimpleFeatureType,
+                         compression: Int = Deflater.NO_COMPRESSION) extends Closeable with Flushable {
```
Comment (jahhulbert-ccri) [inline geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroDataFileWriter.scala]: comment below about default avro compression

File: geomesa-tools/README.md (hunk 2 above)
Comment (elahrvivaz) [inline geomesa-tools/README.md]: will do

Comment (jahhulbert-ccri) [inline geomesa-tools/README.md]: batching is good like you mentioned in chat...later is OK...I imagine the ability to specify 64 or 128 MB blocks for HDFS would be useful

Comment (jahhulbert-ccri) [inline geomesa-tools/README.md]: i'm guessing also the default should be set in the AvroDataFile constructor so people don't have to think about it

Comment (elahrvivaz) [inline geomesa-tools/README.md]: updated default to use compression

**Comment by elahrvivaz** [comment]
This covers GEOMESA-1129 now too

**Comment by jahhulbert-ccri** [comment]
looks pretty good. waiting on stacked commit then will merge

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/ExportCommand.scala [hunk 4]
```diff
@@ -27,25 +28,27 @@ import org.opengis.filter.Filter
...
 
 class ExportCommand(parent: JCommander) extends CommandWithCatalog(parent) with LazyLogging {
+
   override val command = "export"
   override val params = new ExportParameters
 
   override def execute() = {
     val fmt = Formats.fromString(params.format)
     val features = getFeatureCollection(fmt)
+    lazy val avroCompression = Option(params.gzip).map(_.toInt).getOrElse(Deflater.DEFAULT_COMPRESSION)
```
Comment (jahhulbert-ccri) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/ExportCommand.scala]: Deflater.DEFAULT_COMPRESSION I think is actually -1 so perhaps Defalter.DEFLATED ?

**Comment by jahhulbert-ccri** [comment]
5570684449cb4ad67b0c655940e4d14d93bb40bf

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/ExportCommand.scala (hunk 4 above)
Comment (elahrvivaz) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/commands/ExportCommand.scala]: yeah, i checked - default == -1, which currently corresponds to level 6, but may change (although seems unlikely)

**Pull request closed without merging**